script.js

```diff
@@ -103,90 +102,86 @@ createMap(50.07, 14.44);
 
 let cache = {};
 async function getData(url){
-    let result = "";
     if(cache[url] !== undefined) return cache[url].value;
-
     await fetch(url)
```

Comment:
Just noticed: when using `then` approach, async/await should not be there - instead, last function of `then` chain returns the required value and `getData` returns whole promise chain like this:
```
function getData(url){
    ...
    return fetch(url)
    .then(...)
    .then(...);
    .then((...) => {
        ...
        return value;
    });
}
```

Reply:
fixed in last commit

Reply:
Should work without `async` now it is a strange mix of async/then... I guess the problem is when returning cached data from `getData`, promise has to be returned as well. Creating promise from a value can be done by `Promise.resolve(value)`.